checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -143,5 +149,75 @@ def _build_sub_graph(self, blocks_to_render: list[int]) -> l_graph.TerraformLoca
+        for k, v in attrs.items():
+            if isinstance(v, list) and len(v) == 1 and isinstance(v[0], str) and COUNT_KEY in v[0]:
+                attrs[k] = attrs[k][0].replace("${" + COUNT_KEY + "}", str(new_value))
+                attrs[k] = attrs[k][0].replace(COUNT_KEY, str(new_value))
+            elif isinstance(v, dict):
+                self._update_count_attributes(v, new_value)
+
+    def _update_foreach_attributes(self, attrs: dict[str, Any], new_value: str, new_key: Optional[str | int] = ''):
+        for k, v in attrs.items():
+            if isinstance(v, list) and len(v) == 1 and isinstance(v[0], str):
+                if EACH_KEY in v[0]:
+                    attrs[k][0] = attrs[k][0].replace("${" + EACH_KEY + "}", str(new_key) or str(new_value))
+                    attrs[k][0] = attrs[k][0].replace(EACH_KEY, str(new_key) or str(new_value))
+                if EACH_VALUE in v[0]:
+                    attrs[k][0] = attrs[k][0].replace("${" + EACH_VALUE + "}", str(new_value))
+                    attrs[k][0] = attrs[k][0].replace(EACH_KEY, str(new_value))
+            elif isinstance(v, dict):
+                self._update_foreach_attributes(v, new_value, new_key)
+
+    def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | str, new_key: Optional[str] = ''):
+        new_resource = deepcopy(main_resource)
+        if isinstance(new_value, int):
+            self._update_count_attributes(new_resource.attributes, new_value)
+        else:
+            self._update_foreach_attributes(new_resource.attributes, new_value, new_key)
+        if new_key:
+            self._add_index_to_block_properties(new_resource, new_key)
+        else:
+            self._add_index_to_block_properties(new_resource, new_value)
+        self.local_graph.vertices.append(new_resource)
+
+    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], main_resource: TerraformBlock) -> None:
+        if isinstance(statement, list):
+            for new_value in statement:
+                self._create_new_resource(main_resource, new_value)
+        if isinstance(statement, dict):
+            for new_key, new_value in statement.items():
+                self._create_new_resource(main_resource, new_value, new_key=new_key)
+
+    def _delete_main_resource(self, block_index_to_statement: FOR_EACH_BLOCK_TYPE) -> None:
+        new_vertices = [
+            block for i, block in enumerate(self.local_graph.vertices) if i not in block_index_to_statement or not block_index_to_statement[i]
+        ]
+        self.local_graph.vertices = new_vertices
+
+    @staticmethod
+    def _add_index_to_block_properties(block: TerraformBlock, idx: str | int) -> None:
+        block_type, block_name = block.name.split('.')
+        block.id = f"{block.id}[{idx}]"
+        block.name = f"{block.name}[{idx}]"
+        address = CustomAttributes.TF_RESOURCE_ADDRESS
```

Comment:
why not use the const itself? it's a bit confusing, I thought it was the address value